src/renderer/api/kube-object.ts

```diff
@@ -84,21 +84,27 @@ export interface EditPrecondtion {
   message: string;
 }
 
-function requiredKeyUnchanged(key: keyof KubeObject): EditPrecondtion {
-  return {
-    check: (original: KubeObject, edited: KubeObject): boolean => (
-      deepEqual(edited[key], original[key], { strict: true })
-    ),
-    message: `${key} has changed`,
-  };
-}
+function requiredKeyUnchanged(key: keyof KubeObject): EditPrecondtion;
+function requiredKeyUnchanged(key: keyof KubeObject, key2?: keyof KubeObject[typeof key]): EditPrecondtion;
```

Comment:
`key: keyof KubeObject` might be not a valid name, so why not just a `string`?

![image](https://user-images.githubusercontent.com/6377066/113300518-ac049380-9306-11eb-8be5-383a556e7f4c.png)

Reply:
Okay I'll just have strings, though I need multiple args because of `metadata`.

Reply:
We can support single key and use string-path with https://lodash.com/docs/4.17.15#get
e.g. `requiredKeyUnchanged("metadata.namespace")`